internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -186,6 +188,49 @@ func (s *Server) HandleGitHubWebHook() http.HandlerFunc {
 	}
 }
 
+func validatePayloadSignature(r *http.Request, wc *server.WebhookConfig) (payload []byte, err error) {
+	var bodyBytes []byte
+
+	// if there are any previous secrets we need to save the body
+	// because every call to ValidatePayload will consume the body
+	if len(wc.PreviousWebhookSecrets) > 0 {
+		bodyBytes, err = getAndDuplicateBody(r)
+		if err != nil {
+			return
+		}
+	}
+
+	payload, err = github.ValidatePayload(r, []byte(wc.WebhookSecret))
+	if err == nil {
+		return
+	}
+
+	for _, prevSecret := range wc.PreviousWebhookSecrets {
+		r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
```

Comment:
bytes.NewBuffer() is supposed to take ownership of its argument; you probably want `bytes.NewReader()` instead.

Reply:
good catch, thanks.

Reply:
fixed!